script.js

```diff
@@ -1,4 +1,381 @@
+// 2 players  - User and dealer (computer)
+// 1 deck of cards
+// starting hand for 2 cards
+// compare winning conditions - win, lose, draw
+
+// Flow of the game
+// The deck is shuffled first
+// User clicks submit to deal the cards
+// Game checks if user has blackjack = auto win
+// Cards are displayed to user
+// User select hit or stand.
+// If hit, deals another card and check for win/lose conditions
+// If stand, dealer will decided to hit or stand based on the rules set (16> stand.)
+// Compare dealer and user number, and conclude winner
+
+// Create a deck
+// Use objects. Card contain suits, number and rank.
+// 52 cards, 13 cards for each suit.
+// Test git push
+
+/// 1. CREATION OF DECKS FUNCTIONS
+
+var makeDeck = function () {
+  var suits = ["Spade", "Heart", "Club", "Diamond"];
+  var deck = [];
+
+  for (var i = 0; i < suits.length; i += 1) {
+    for (var j = 1; j <= 13; j += 1) {
+      var cardName = assignCardNames(j);
+      var cardNum = assignCardNumbers(j);
+      if (cardName == "ace") {
+        cardNum = 11;
+      }
+      //console.log(cardname);
+      var card = {
+        name: cardName,
+        suit: suits[i],
+        number: cardNum,
+        rank: j,
+      };
+      //console.log(card);
+      deck.push(card);
+    }
+  }
+  return deck;
+};
+
+// Assign Card Names (ace, jack, queen,king)
+// insert rank into input
+var assignCardNames = function (input) {
+  if (input == 1) {
+    return "ace";
+  } else if (input == 11) {
+    return "jack";
+  } else if (input == 12) {
+    return "queen";
+  } else if (input == 13) {
+    return "king";
+  } else {
+    return String(input);
+  }
+};
+
+// Assign Card Numbers (1,11,12,13 -> Number 1 and 10)
+var assignCardNumbers = function (input) {
+  if (input == 1) {
+    return 1;
+  } else if (input == 11) {
+    return 10;
+  } else if (input == 12) {
+    return 10;
+  } else if (input == 13) {
+    return 10;
+  } else {
+    return input;
+  }
+};
+
+// Shuffle a deck - Inputs: Deck (array type)
+var shuffleDeck = function (input) {
+  var i = 0;
+  while (i < input.length) {
+    var randomNumber = randomNum();
+    var proxycard_i = deck[i];
+    input[i] = input[randomNumber];
+    input[randomNumber] = proxycard_i;
+    i += 1;
+  }
+  return input;
+};
+
+// Get a random number
+var randomNum = function () {
+  return Math.floor(Math.random() * 52);
+};
+
+// GAME BEGINS HERE
+
+var gamestate = "START_STATE"; // START_STATE, HIT_STATE, STAND_STATE
+var playerHand = [];
+var computerHand = [];
+var deck = makeDeck();
+var shuffledDeck = shuffleDeck(deck);
+var playerHandValue = 0;
+var computerHandValue = 0;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  //check gamestate
+  if (checkCorrectInput(input)) {
```

Comment:
For functions that return booleans, it'd be more ideal if your function name resolves to a yes or no. i.e. isCorrectInput(input)